runtime/benches/append_vec.rs

```diff
@@ -237,10 +234,7 @@ fn bench_account_serialize_bincode(bencher: &mut Bencher) {
     });
 
     // make sure compiler doesn't delete the code.
-    let index = thread_rng().gen_range(0, len);
-    if memory[index] != 0 {
-        println!("memory: {}", memory[index]);
-    }
+    let index = test::black_box(thread_rng().gen_range(0, num));
```

Comment:
this doesn't seem right.. you don't need the random index if you are using black_box, and shouldn't the black_box be around the code which is in danger of being optimized like the `serialize_into` in the `iter(||` block?

Reply:
Well I guess deserializing a random index might still be a good idea, but seems that black_box is still not in the right spot.

Reply:
Around `memory` maybe? The comment isn't too clear about what would otherwise be optimized out. I'm just trying to get rid of the println without breaking the benchmark.

Reply:
It's so the compiler doesn't optimize out the serialize_into call. There are no side-affects of doing that except changing memory.